### ISSUE TITLE
some questions about caffe deconvolution

### ISSUE THREAD

**Issue by jsjs0827** [chengyangfu/caffe#5]
I operate deconv on my networks and show the feature maps，I find that the feature maps are quite white（maybe all value>0），did caffe uniformize the data before feed them to deconvolutional layer? Where can I change the mechanism？

我在我自己的网络上做反卷积，但是得到的特征图都偏白，caffe在做反卷积的时候要归一化的吗？我在源码哪能改掉这个机制？

**Comment by chengyangfu** [comment]
Hi @jsjs0827 
For the first question, I don't think deconvolutional layer will change preprocessing the input before operating it. 
Did you check the feature maps before feeding to deconvolutional layers? The feature maps before and after deconvolutional layers should be similar but in different resolutions. 

May I ask what's your problem? More information could lead to better suggestion. 
According to my experience, I will suggest to use simple bilinear upsampling to replace the learned deconvoltuional layers in detection. If you are working on high resolution images, maybe the learned deconvoltuinoal layer can give better performance.

**Comment by jsjs0827** [comment]
So appreciated for your reply.
About the first question,I would check it myself right now.
I do the experiments in caffe,the deconv layer used is the inherent layer in caffe,should I follow a relu after the deconv layer?Which is the better combination for the deconv?

**Comment by chengyangfu** [comment]
I didn't use the ReLU. You can try different values for the parameter "group". I think it will be helpful.  
Another thing I found is useful is adding 3x3 conv filter and BN after deconvolution. 

Efficient : bilinear upsampling directly. 
Better way : Deconvolution  + 3x3conv + BN

**Comment by jsjs0827** [comment]
Yeah, I used the combination myself before,but the feature map I get is not very good for the detection.You use the deconv for what,specific task?
